app/controlplane/pkg/biz/group.go

```diff
@@ -0,0 +1,266 @@
+//
+// Copyright 2025 The Chainloop Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package biz
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/chainloop-dev/chainloop/app/controlplane/pkg/auditor/events"
+	"github.com/chainloop-dev/chainloop/app/controlplane/pkg/pagination"
+
+	"github.com/go-kratos/kratos/v2/log"
+	"github.com/google/uuid"
+)
+
+type GroupRepo interface {
+	// List retrieves a list of groups in the organization, optionally filtered by name, description, and owner.
+	List(ctx context.Context, orgID uuid.UUID, filterOpts *ListGroupOpts, paginationOpts *pagination.OffsetPaginationOpts) ([]*Group, int, error)
+	// Create creates a new group.
+	Create(ctx context.Context, orgID uuid.UUID, opts *CreateGroupOpts) (*Group, error)
+	// Update updates an existing group.
+	Update(ctx context.Context, orgID uuid.UUID, groupID uuid.UUID, opts *UpdateGroupOpts) (*Group, error)
+	// FindByOrgAndID finds a group by its organization ID and group ID.
+	FindByOrgAndID(ctx context.Context, orgID uuid.UUID, groupID uuid.UUID) (*Group, error)
+	// SoftDelete soft-deletes a group by marking it as deleted.
+	SoftDelete(ctx context.Context, orgID uuid.UUID, groupID uuid.UUID) error
+	// ListMembers retrieves a list of members in a group, optionally filtered by maintainer status.
+	ListMembers(ctx context.Context, orgID uuid.UUID, groupID uuid.UUID, opts *ListMembersOpts, paginationOpts *pagination.OffsetPaginationOpts) ([]*GroupMembership, int, error)
```

Comment:
I am a little bit confused, is there any way in this PR to add members to a group?